### ISSUE TITLE
Scripting addition failed to inject payload into Dock.app

### ISSUE THREAD

**Issue by sidv23** [koekeishiya/yabai#1864]
Hi, I'm running into an issue when running `yabai` on my M2 Mac Mini. (I have `yabai` setup with SIP disabled on my M1 MBP and an Intel MBP successfully in the past) Any help would be greatly appreciated! 

---

### Issue

Scripting addition fails to load

```sh
▶ sudo yabai --load-sa
could not retrieve task port for pid: 1053
yabai: scripting-addition failed to inject payload into Dock.app!
```

The process with `pid 1053` is `Dock.app`

```sh
▶ ps aux | grep 1053
<username>       1053   0.0  0.4 409716608  35216   ??  S     5:41PM   0:01.89 /System/Library/CoreServices/Dock.app/Contents/MacOS/Dock
```

---

### How I got here

1. I installed `yabai` (from HEAD) using the instructions in the wiki [here](https://github.com/koekeishiya/yabai/wiki/Installing-yabai-(from-HEAD))
```sh
▶ yabai --version
yabai-v5.0.8
```

2. disabled SIP on my machine using the instructions [here](https://github.com/koekeishiya/yabai/wiki/Disabling-System-Integrity-Protection)
```sh
▶ csrutil status
System Integrity Protection status: unknown (Custom Configuration).

Configuration:
	Apple Internal: disabled
	Kext Signing: enabled
	Filesystem Protections: disabled
	Debugging Restrictions: disabled
	DTrace Restrictions: enabled
	NVRAM Protections: disabled
	BaseSystem Verification: enabled
	Boot-arg Restrictions: disabled
	Kernel Integrity Protections: enabled
	Authenticated Root Requirement: enabled
```

3. configured the scripting addition using the instructions [here](https://github.com/koekeishiya/yabai/wiki/Installing-yabai-(from-HEAD)#configure-scripting-addition)
```sh
▶ sudo head /private/etc/sudoers.d/yabai
<ussername> ALL=(root) NOPASSWD: sha256:0c...b3 /opt/homebrew/bin/yabai --load-sa
```

4. Configured the `boot-args` 
```sh
▶ nvram -p | grep boot-args
boot-args	-arm64e_preview_abi
```

---

### What I've tried

1. Uninstall `yabai` -> Reboot -> Reinstall `yabai`
2. Some solutions mentioned in the comments of the following issues:
  - #725
  - #1158 
  - #1234

---

### Some more logs

```sh 
▶ file $(which yabai)
/opt/homebrew/bin/yabai: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64:Mach-O 64-bit executable arm64]
/opt/homebrew/bin/yabai (for architecture x86_64):	Mach-O 64-bit executable x86_64
/opt/homebrew/bin/yabai (for architecture arm64):	Mach-O 64-bit executable arm64


▶ cat /Library/ScriptingAdditions/yabai.osax/Contents/Info.plist | grep -A 1 CFBundleVersion
<key>CFBundleVersion</key>
<string>2.0.18</string>


▶ file /Library/ScriptingAdditions/yabai.osax/Contents/Resources/payload.bundle/Contents/MacOS/payload
/Library/ScriptingAdditions/yabai.osax/Contents/Resources/payload.bundle/Contents/MacOS/payload: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit dynamically linked shared library x86_64] [arm64e:Mach-O 64-bit dynamically linked shared library arm64e]
/Library/ScriptingAdditions/yabai.osax/Contents/Resources/payload.bundle/Contents/MacOS/payload (for architecture x86_64):	Mach-O 64-bit dynamically linked shared library x86_64
/Library/ScriptingAdditions/yabai.osax/Contents/Resources/payload.bundle/Contents/MacOS/payload (for architecture arm64e):	Mach-O 64-bit dynamically linked shared library arm64e
```

---

Any help on fixing this and getting the scripting addition to work on my machine would be much appreciated. Thanks!

**Comment by milan-panta** [comment]
Having the same issue on my M2 Macbook Pro running yabai 5.0.8 on Sonoma.

**Comment by fr0zn** [comment]
Same issue on the M2 Macbook Pro, running yabai 5.0.8 on Sonoma. Appart from what is already mentioned, i am getting the following notification:

```
payload (0x0) doesn't support this macOS version
```

**Comment by sidv23** [comment]
The issue was fixed on my end when I upgraded to `yabai v5.0.9`. I uninstalled and reinstalled the payload to `payload v2.0.21`.
